packages/web/hooks/use-swap.ts

```diff
@@ -372,12 +308,15 @@ export function useSwap({
         inAmountInput.reset();
```

Comment:
If the promise resolves or rejects quickly, it's possible that inAmountInput.reset() could be called twice, leading to unexpected behavior. Move the inAmountInput.reset() call to the finally block to ensure it's only called once, regardless of the promise's outcome.